examples/cookie-game/src/index.ts

```diff
@@ -0,0 +1,336 @@
+
+const printMessageWithSource = (message: string, source: transaction) => {
+    console.log(message);
+    console.log(source);
+}
+
+const saveState = (source: transaction, sourceValue: cookieBakerType) => {
+    printMessageWithSource("Saving state", sourceValue);
+    set(source, sourceValue);
+    console.log("Successfully saved state");
+}
+
+const transition = (tx: transaction) => {
+    // source -> tz1 address
+    // op_hash / tx_hash => BLAKE2B => resolved as string
+    // operation => any
+    const source = tx.source;
+    const operation = tx.operation;
+    console.log("Getting source");
+    const sourceValue = JSON.parse(get(source));
+    const cookieBaker: cookieBakerType = createCookieBaker(
+        sourceValue.cookieBaker.numberOfCookie,
+        sourceValue.cookieBaker.numberOfCursor,
+        sourceValue.cookieBaker.numberOfGrandma,
+        sourceValue.cookieBaker.numberOfFarm,
+        sourceValue.cookieBaker.numberOfMine,
+        sourceValue.cookieBaker.numberOfFactory,
+        sourceValue.cookieBaker.numberOfBank,
+
+        sourceValue.cookieBaker.numberOfTemple,
+        sourceValue.cookieBaker.numberOfWizard,
+        sourceValue.cookieBaker.numberOfShipment,
+        sourceValue.cookieBaker.numberOfAlchemy,
+        sourceValue.cookieBaker.numberOfPortal,
+        sourceValue.cookieBaker.numberOfTimeMachine,
+        sourceValue.cookieBaker.numberOfAntimatter,
+        sourceValue.cookieBaker.numberOfPrism,
+        sourceValue.cookieBaker.numberOfChanceMaker,
+        sourceValue.cookieBaker.numberOfFractal,
+        sourceValue.cookieBaker.numberOfJavaScript,
+        sourceValue.cookieBaker.numberOfIdleverse,
+        sourceValue.cookieBaker.numberOfCordex,
+
+        sourceValue.cookieBaker.numberOfFreeCursor,
+        sourceValue.cookieBaker.numberOfFreeGrandma,
+        sourceValue.cookieBaker.numberOfFreeFarm,
+        sourceValue.cookieBaker.numberOfFreeMine,
+        sourceValue.cookieBaker.numberOfFreeFactory,
+        sourceValue.cookieBaker.numberOfFreeBank,
+        sourceValue.cookieBaker.numberOfFreeTemple,
+        sourceValue.cookieBaker.numberOfFreeWizard,
+        sourceValue.cookieBaker.numberOfFreeShipment,
+        sourceValue.cookieBaker.numberOfFreeAlchemy,
+        sourceValue.cookieBaker.numberOfFreePortal,
+        sourceValue.cookieBaker.numberOfFreeTimeMachine,
+        sourceValue.cookieBaker.numberOfFreeAntimatter,
+        sourceValue.cookieBaker.numberOfFreePrism,
+        sourceValue.cookieBaker.numberOfFreeChanceMaker,
+        sourceValue.cookieBaker.numberOfFreeFractal,
+        sourceValue.cookieBaker.numberOfFreeJavaScript,
+        sourceValue.cookieBaker.numberOfFreeIdleverse,
+        sourceValue.cookieBaker.numberOfFreeCordex,
+    );
+
+
+    switch (operation) {
+        case actions.incrementCookie: {
+            const updatedCookieBaker = addCookie(cookieBaker);
+            //update state
+            sourceValue.cookieBaker = updatedCookieBaker;
```

Comment:
I guess we can do this, it will be better and easier to read

```suggestion
            sourceValue.cookieBaker = addCookie(cookieBaker);
```